### PR TITLE
admins can add specific fantasy affixes to items

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Added "Add Fantasy Affix" vv dropdown

## Why It's Good For The Game

I keep having recurring good ideas for how to make special adminbuse weapons from these really cool prefixes but I don't have the power to do so... so here you go!

This is a lot like when i added adding components to the VV dropdown and that was a historic day for new levels of buse

## Changelog
:cl:
admin: Added fantasy affixes to the VV dropdown for items!
/:cl:

### CODE REVIEW

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -364,6 +364,59 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
+			fantasy_quality++
+		else
+			fantasy_quality--
+
+		//Apply fantasy with affix. failing this should never happen, but if it does it should not be silent.
+		if(AddComponent(/datum/component/fantasy, fantasy_quality, list(affix), FALSE, FALSE) == COMPONENT_INCOMPATIBLE)
+			to_chat(usr, "<span class='warning'>Fantasy component not compatible with [src].</span>")
+			CRASH("fantasy component incompatible with object of type: [type]")
+
+		to_chat(usr, "<span class='notice'>[src] now has [picked_affix_name]!</span>")
```
Comment (coiax) [inline code/game/objects/items.dm:416]: Won't this already have the new affix as part of the name?

Something like

`the claymore of lizard smiting now has bane of some random mob`?

File: code/datums/components/fantasy/suffixes.dm [hunk 2]
```diff
@@ -47,6 +48,7 @@
 
 //////////// Good suffixes
 /datum/fantasy_affix/bane
+	name = "bane of some random mob"
```
Comment (coiax) [inline code/datums/components/fantasy/suffixes.dm:51]: I think I'd prefer these names to more accurately reflect the names they apply to the item.

`of <mobtype> slaying (random species, carbon or simple animal)`

File: code/datums/components/fantasy/suffixes.dm [hunk 3]
```diff
@@ -149,6 +153,7 @@
 	return "[newName] of [initial(picked_projectiletype.name)] shrapnel"
 
 /datum/fantasy_affix/strength
+	name = "knockback (strength in game)"
```
Comment (coiax) [inline code/datums/components/fantasy/suffixes.dm:156]: `of strength (knockback)`

File: code/datums/components/fantasy/suffixes.dm [hunk 4]
```diff
@@ -161,6 +166,7 @@
 //////////// Bad suffixes
 
 /datum/fantasy_affix/fool
+	name = "clown honks (fool in game)"
```
Comment (coiax) [inline code/datums/components/fantasy/suffixes.dm:169]: `of the fool (honking)`

File: code/game/objects/items.dm [hunk 5]
```diff
@@ -364,6 +364,59 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
+		affixes.Remove(affix)
+		QDEL_LIST_ASSOC(affixes) //remove the rest, we didn't use them
+		var/fantasy_quality = 0
+		if(affix.alignment & AFFIX_GOOD)
+			fantasy_quality++
+		else
+			fantasy_quality--
+
+		//Apply fantasy with affix. failing this should never happen, but if it does it should not be silent.
+		if(AddComponent(/datum/component/fantasy, fantasy_quality, list(affix), FALSE, FALSE) == COMPONENT_INCOMPATIBLE)
```
Comment (coiax) [inline code/game/objects/items.dm:412]: I believe that AddComponent, being a macro hack, doesn't have named arguments, so could you move these arguments out as variables?

In this case

`var/canFail = FALSE`
`var/announce = FALSE`

**Comment by tralezab** [comment]
Integration test seemed spontaneous, this is good to go 👍

**Pull request merged**